modules/integration/tests-integration/bpmn/src/test/java/org/wso2/bps/integration/tests/bpmn/JavaServiceTaskDeploymentUndeploymentTestCase.java

```diff
@@ -0,0 +1,104 @@
+
+    public void deployUnDeployJavaServiceTaskBPMNPackage() throws Exception {
+        init();
+        ActivitiRestClient tester = new ActivitiRestClient(bpsServer.getInstance().getPorts().get("http")
+                , bpsServer.getInstance().getHosts().get("default"));
+
+        String filePath = FrameworkPathUtil.getSystemResourceLocation() + File.separator
+                          + BPMNTestConstants.DIR_ARTIFACTS + File.separator
+                          + BPMNTestConstants.DIR_BPMN + File.separator + "sampleJavaServiceTask.bar";
+
+        String fileName = "sampleJavaServiceTask.bar";
+        String[] deploymentResponse;
+
+        deploymentResponse = tester.deployBPMNPackage(filePath, fileName);
+        Assert.assertTrue("Deployment Successful", deploymentResponse[0].contains(BPMNTestConstants.CREATED));
+        String[] deploymentCheckResponse = tester.getDeploymentInfoById(deploymentResponse[1]);
+        Assert.assertTrue("Deployment Present", deploymentCheckResponse[2].contains(fileName));
+
+
+        String[] definitionResponse = tester.findProcessDefinitionInfoById(deploymentResponse[1]);
+        Assert.assertTrue("Search Success", definitionResponse[0].contains("200"));
+
+        //Starting and Verifying Process Instance
+        String[] processInstanceResponse = tester.startProcessInstanceByDefintionID(definitionResponse[1]);
+        Assert.assertTrue("Process Instance Started", processInstanceResponse[0].contains(BPMNTestConstants.CREATED));
+        String searchResponse = tester.searchProcessInstanceByDefintionID(definitionResponse[1]);
+        Assert.assertTrue("Process Instance Present", searchResponse.contains(BPMNTestConstants.OK));
+
+        //check for varible value if true
+        String[] innovationResponse = tester.getValueOfVariableOfProcessInstanceById(processInstanceResponse[1], "executionState");
+        Assert.assertTrue("Variable Present", innovationResponse[0].contains(BPMNTestConstants.OK));
+        Assert.assertTrue("Variable Present", innovationResponse[1].contains("executionState"));
+        Assert.assertTrue("Variable Value is True", innovationResponse[2].contains("true"));
+
+
+        //Suspending the Process Instance
+        String[] suspendResponse = tester.suspendProcessInstanceById(processInstanceResponse[1]);
+        Assert.assertTrue("Process Instance has been suspended", suspendResponse[0].contains(BPMNTestConstants.OK));
+        Assert.assertTrue("Process Instance has been suspended", suspendResponse[1].contains("true"));
+
+        //Deleting a Process Instance
+        String deleteStatus = tester.deleteProcessInstanceByID(processInstanceResponse[1]);
+        Assert.assertTrue("Process Instance Removed", deleteStatus.contains(BPMNTestConstants.NO_CONTENT));
+
+        String status = tester.unDeployBPMNPackage(deploymentResponse[1]);
+        Assert.assertTrue("Package UnDeployed", status.contains(BPMNTestConstants.NO_CONTENT));
+    }
```

Comment:
Created new methods to get all list of  deployment/process instance and query for the  deployment/process instance to ensure that the deployment/process instance has been removed. Fixed in the latest commit.